Exiled.Events/Patches/Events/Player/Shoot.cs

```diff
@@ -29,276 +37,101 @@ namespace Exiled.Events.Patches.Events.Player
     [HarmonyPatch(typeof(WeaponManager), nameof(WeaponManager.CallCmdShoot))]
     internal static class Shoot
     {
-        private static bool Prefix(WeaponManager __instance, GameObject target, HitBoxType hitboxType, Vector3 dir, Vector3 sourcePos, Vector3 targetPos)
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
         {
-            try
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            // search for "this._hub" before the only "PlayerEffect.ServerDisable()"
+            int offset = 17;
+            int index = newInstructions.FindLastIndex(instruction => instruction.opcode == OpCodes.Callvirt && (MethodInfo)instruction.operand == Method(typeof(PlayerEffect), nameof(PlayerEffect.ServerDisable))) - offset;
+
+            LocalBuilder ev1 = generator.DeclareLocal(typeof(ShootingEventArgs));
```

Comment:
You should call them shootingEv and shotEv, to understand them better when looking at the IL code.